torch/_inductor/sizevars.py

```diff
@@ -341,9 +342,9 @@ def evaluate_min(self, left: Expr, right: Expr) -> Expr:
         """return the smaller of left and right, and guard on that choice"""
         lv = self.size_hint(left)
         rv = self.size_hint(right)
-        if lv == rv:
+        if self.shape_env._maybe_evaluate_static(sympy.Eq(lv, rv)):
```

Comment:
This is wrong, this will return None if it can't evaluate, but that doesn't mean that lv and rv aren't equal. Here, you just end up wildly (incorrectly) assuming right < left